src/core/provider.h

```diff
@@ -178,16 +178,23 @@ class InferRequestProvider {
       const std::string& name, const void** content, size_t* content_byte_size,
       TRTSERVER_Memory_Type* memory_type, int64_t* memory_type_id);
 
-  // Retrieve the data buffer of input 'name'.
+  // Retrieve the data buffer of input 'name'. This function will not check
+  // input override.
   Status GetMemory(
       const std::string& name, std::shared_ptr<Memory>* input_buffer);
 
+  // Similar to above, but the function caller does not own the Memory object,
+  // nor extend its lifetime. This function will check input override.
+  Status GetMemory(const std::string& name, const Memory** input_buffer);
```

Comment:
Can we name GetMemoryWithOverride or similar since otherwise it is somewhat confusing with the other GetMemory